fastlane/Fastfile

```diff
@@ -593,6 +593,37 @@ end
       SimCtl.create_device device, devicetype, runtime
     }
   end
+
+  #####################################################################################
+  # UI tests lanes
+  #####################################################################################
+
+    # Props to work done by others in this file and in
+    # https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/fastlane/Fastfile
+    # this is a combined rip-off from both
+
+    #####################################################################################
+    # run_UI_tests
+    # -----------------------------------------------------------------------------------
+    # Usage:
+    # bundle exec fastlane run_UI_tests [device:<Name of the iOS Simulator >]
+    #
+    # Example:
+    # bundle exec fastlane run_UI_tests device:"iPhone 12"
+    #####################################################################################
+  # A lane to run a subset of automated UI tests. Since only login tests are
+  # selected for now, there will be no credentials race condition.
+  desc "Run UI tests"
+  lane :run_UI_tests do | options |
+  scan(
+    workspace: workspace_name,
+    scheme: ui_tests_scheme,
+    device: options[:device],
+    prelaunch_simulator: true,
+    reset_simulator: true,
+    only_testing: "SimplenoteUITests/SimplenoteUISmokeTestsLogin"
+  )
```

Comment:
When I run the lane locally (simply with `bundle exec fastlane run_UI_tests`, it failed with 

```
Reason: Cannot test target “SimplenoteUITests” on “iPhone 11”: iPhone 11’s iOS Simulator 13.4 doesn’t match SimplenoteUITests’s iOS Simulator 14.4 deployment target.
```

Even though this lane is only meant to run on CI, really, it's necessary for it to run locally out of the box, too.

We may want to use fallback values for both device and iOS version:

```suggestion
  scan(
    workspace: workspace_name,
    scheme: ui_tests_scheme,
    device: options[:device] || 'iPhone 12',
    deployment_target_version: options[:ios_version] || '14.4'
    prelaunch_simulator: true,
    reset_simulator: true,
    only_testing: "SimplenoteUITests/SimplenoteUISmokeTestsLogin"
  )
```

(There's actually a `simulator_version` parameter at the bottom of the file that we could use instead of `14.4`)

Alternatively, we could check that the values for the device and iOS version are explicitly passed to the lane before calling `scan`. 

I haven't tested it, but something like:


```ruby
usage = "Usage bundle exec fastlane run_ui_tests device:<device> ios_version:<ios version>"

UI.user_error!(usage) if options[:device].nil? || options[:ios_version].nil?

scan(...)
```

Reply:
@mokagio Added the fallback for `device`, thank you! As for `deployment_target_version` parameter: if I understood correctly from the [doc](https://docs.fastlane.tools/actions/run_tests/), this does not specify the iOS version that will be actually used, but rather filters out the devices that do not run the specified iOS version, if the device was not specified. Given that we specify the device, it looks like it adds no value? Maybe I got if wrong 🤔

Reply:
I'll do more testing with this locally now...

Reply:
It actually confuses me even more now, because the Xcode target specifies the `iOS Deployment Target` anyway already, which I guess it the same thing as `deployment_target_version`

Reply:
Oh... Interesting... 🤔 

Maybe then there's no direct way to tell `scan` which simulator version to use. I did a quick Google and found [this](https://github.com/fastlane/fastlane/issues/10338) and [this](https://github.com/fastlane/fastlane/issues/14953) issues which seem to confirm that.

Let's not worry about that for now then. Using the latest version of Xcode in CI should guarantee the use of the latest version of iOS in the Simulator and that might be enough for the moment. We'll look into more details if we'll ever need to run across different version of iOS, I guess 🤷‍♂️